tests/test_tenx.py

```diff
@@ -0,0 +1,125 @@
+from . import sourmash_tst_utils as utils
+import sourmash.tenx as sourmash_tenx
+
+import tempfile
+import pysam as bs
+
+
+def test_read_barcodes_file():
+    filename = utils.get_test_data('10x-example/barcodes.tsv')
+    barcodes = sourmash_tenx.read_barcodes_file(filename)
+    assert len(barcodes) == 10
+
+
+def test_read_bam_file():
+    filename = utils.get_test_data('10x-example/possorted_genome_bam.bam')
+    bam_file = sourmash_tenx.read_bam_file(filename)
+    assert isinstance(bam_file, bs.AlignmentFile)
+    total_alignments = sum(1 for _ in bam_file)
+    assert total_alignments == 1714
+
+
+def test_shard_bam_file():
+    filename = utils.get_test_data('10x-example/possorted_genome_bam.bam')
+    bam_file = sourmash_tenx.read_bam_file(filename)
+    assert isinstance(bam_file, bs.AlignmentFile)
+
+    expected_alignments = sum(1 for _ in bam_file)
+    bam_shard_files = sourmash_tenx.shard_bam_file(
+        filename, expected_alignments, tempfile.mkdtemp())
```

Comment:
if you look at some of the other tests, there's the code:

```
with utils.TempDirectory() as location
```
that I think you should be using here. That way the test framework will clean up after you :). But this may be better resolved in the shard_bam_file function...

Reply:
changed it